### PR TITLE
add pre-commit checks and checks summary

### PR DESCRIPTION
**Pull request by giovp**

### CODE REVIEW

**Comment by grst** [comment]
Come have some beers!

File: {{cookiecutter.project_name}}/.pre-commit-config.yaml [hunk 1]
```diff
@@ -1,23 +1,82 @@
+fail_fast: false
+default_language_version:
+  python: python3
+default_stages:
+  - commit
+  - push
+minimum_pre_commit_version: 2.9.3
```
Comment (ivirshup) [inline {{cookiecutter.project_name}}/.pre-commit-config.yaml:7]: Update this?

@michalk8 says it's at 2.16 at least

```suggestion
minimum_pre_commit_version: 2.16.0
```

File: {{cookiecutter.project_name}}/.pre-commit-config.yaml [hunk 2]
```diff
@@ -1,23 +1,82 @@
...
+    hooks:
+      - id: yesqa
+        additional_dependencies:
+          [
+            flake8-tidy-imports,
+            flake8-docstrings,
+            flake8-rst-docstrings,
+            flake8-comprehensions,
+            flake8-bugbear,
+            flake8-blind-except,
```
Comment (ivirshup) [inline {{cookiecutter.project_name}}/.pre-commit-config.yaml:41]: Can these go in the .flake8 file?

Comment (ivirshup) [reply {{cookiecutter.project_name}}/.pre-commit-config.yaml:41]: If we add them there, do these have to be dev dependencies? Probably don't want that.

Comment (giovp) [reply {{cookiecutter.project_name}}/.pre-commit-config.yaml]: they can't go in `.flake8` apparently, only local plugin.

**Review by flying-sheep — APPROVED** [review]
I think we should merge this before more conflicts happen

**Comment by giovp** [comment]
yeah we should wait for #30 and then merge that

**Comment by giovp** [comment]
so pre-commit are failing locally, I'll have a look later

**Comment by giovp** [comment]
I was still working on it 😅

**Pull request merged**